app/callbacks/views/sns.py

```diff
@@ -0,0 +1,416 @@
+                        sys.exc_info()[1]
+                    )
+                ),
+                log_level=logging.DEBUG,
+            ) as log_context_clamd:
+                # hypothetically there is a race condition between the time of scanning the file and fetching the
+                # version here when freshclam could give us a new definition file, making this information incorrect,
+                # but it's a very small possibility
+                clamd_version = clamd_client.version()
+                log_context_clamd.update({"clamd_version": clamd_version})
+
+            # keep in mind we only have 256 chars to play with here, so keep it brief
+            new_av_status = {
+                "result": "pass" if clamd_result[0] == "OK" else "fail",
+                "clamdVerStr": clamd_version,
+                "ts": datetime.datetime.utcnow().isoformat(),
+            }
+            new_av_status_json = json.dumps(new_av_status, separators=(',', ':'))
+
+            # Now we briefly re-check the object's tags to ensure they weren't set by something else while we were
+            # scanning. Note the impossibility of avoiding all possible race conditions as S3's API doesn't allow any
+            # form of locking. What we *can* do is make the possible time period between check-tags and set-tags as
+            # small as possible...
+            with log_external_request(
+                "S3",
+                "get object tagging [{s3_bucket_name}/{s3_object_key} versionId {s3_object_version}]",
+                logger=current_app.logger,
+            ) as log_context_s3:
+                log_context_s3.update(base_log_context)
+                tagging_tag_set = s3_client.get_object_tagging(
+                    Bucket=s3_bucket_name,
+                    Key=s3_object_key,
+                    VersionId=s3_object_version,
+                )["TagSet"]
+
+            av_status_json = _tag_value_from_tag_set(tagging_tag_set, "avStatus")
+
+            if av_status_json is not None:
+                current_app.logger.warning(
+                    "Object was tagged with new 'avStatus' ({existing_av_status}) while we were scanning. "
+                    "Not applying our own 'avStatus' result ({unapplied_av_status})",
+                    extra={
+                        "existing_av_status": av_status_json,
+                        "unapplied_av_status": new_av_status_json,
+                    },
+                )
+                continue
+
+            tagging_tag_set = _tag_set_updated_with_value(tagging_tag_set, "avStatus", new_av_status_json)
+
+            with log_external_request(
+                "S3",
+                "put object tagging [{s3_bucket_name}/{s3_object_key} versionId {s3_object_version}]",
+                logger=current_app.logger,
+            ) as log_context_s3:
+                log_context_s3.update(base_log_context)
+                tagging_tag_set = s3_client.put_object_tagging(
```

Comment:
I don't think you need to reassign `tagging_tag_set` here - it's not used again.